src/compiler/checker.ts

```diff
@@ -12701,6 +12701,19 @@ namespace ts {
                     return anyType;
                 }
                 if (accessExpression && !isConstEnumObjectType(objectType)) {
+                    if (isObjectLiteralType(objectType)) {
+                        if (indexType.flags & (TypeFlags.StringLiteral | TypeFlags.NumberLiteral)) {
+                            diagnostics.add(createDiagnosticForNode(accessExpression, Diagnostics.Property_0_does_not_exist_on_type_1, (indexType as StringLiteralType).value, typeToString(objectType)));

```

Comment:
Hm, looking at the baselines, we may only be safe to add this error (and change the result to `undefined` rather than `any` or `error`) if `noImplicitAny` is set; otherwise we'd be breaking some longstanding `any`ing behavior in our loose checking mode.

Reply:
https://github.com/microsoft/TypeScript/pull/37903/commits/91d29da6fe16061e287dde9fb7bbb83bac12e121
I fixed this ↑, does it make sense?